File: backend/src/domain/ports/game.port.ts
import { Game } from '../entities/game';
import { GameId } from '../value-objects/game-id';

export interface GamePort {
  save(game: Game): Promise<Game>;
  findAll(): Promise<Game[]>;
  findById(gameId: GameId): Promise<Game | null>;
  findByIds(gameIds: GameId[]): Promise<Game[]>;
  delete(gameId: GameId): Promise<boolean>;
}

File: backend-go/internal/domain/ports/game_port.go
package ports

import (
	"context"

	"github.com/posiposi/dragons-counter/backend-go/internal/domain/game"
)

// GameQueryPort は試合データの読み取り操作を定義するポートインターフェースである。
type GameQueryPort interface {
	FindAll(ctx context.Context) ([]game.Game, error)
	FindByID(ctx context.Context, id game.GameID) (*game.Game, error)
	FindByIDs(ctx context.Context, ids []game.GameID) ([]game.Game, error)
}

// GameCommandPort は試合データの書き込み操作を定義するポートインターフェースである。
type GameCommandPort interface {
	Save(ctx context.Context, g game.Game) error
	Delete(ctx context.Context, id game.GameID) (bool, error)
}
